server/src/main/java/org/jboss/as/server/mgmt/UndertowHttpManagementService.java

```diff
@@ -237,17 +266,17 @@ public synchronized void start(StartContext context) throws StartException {
         InetSocketAddress bindAddress = null;
         InetSocketAddress secureBindAddress = null;
 
-        final SocketBinding basicBinding = injectedSocketBindingValue.getOptionalValue();
-        final SocketBinding secureBinding = injectedSecureSocketBindingValue.getOptionalValue();
-        final NetworkInterfaceBinding interfaceBinding = interfaceBindingValue.getOptionalValue();
-        final NetworkInterfaceBinding secureInterfaceBinding = secureInterfaceBindingValue.getOptionalValue();
+        final SocketBinding basicBinding = socketBindingSupplier != null ? socketBindingSupplier.get() : null;
+        final SocketBinding secureBinding = secureSocketBindingSupplier != null ? secureSocketBindingSupplier.get() : null;
+        final NetworkInterfaceBinding interfaceBinding = interfaceBindingSupplier != null ? interfaceBindingSupplier.get() : null;
+        final NetworkInterfaceBinding secureInterfaceBinding = secureInterfaceBindingSupplier != null ? secureInterfaceBindingSupplier.get() : null;
         if (interfaceBinding != null) {
             useUnmanagedBindings = true;
-            final int port = portValue.getOptionalValue();
+            final int port = this.port != null ? this.port : -1;
```

Comment:
I think this null check can be changed to an assert this.port != null.  If an interfaceBinding is provided a port must be provided, and the intended user if it, org.jboss.as.host.controller.operations.HttpManagementAddHandler always provides a port.

Reply:
Updated accordingly @bstansberry